src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SmkRulesAndCheckpointsCompletionContributor.kt

```diff
@@ -132,34 +151,24 @@ private class SmkRulesAndCheckpointsObjectsCompletionProvider : CompletionProvid
 
 // the following functions can be used to implement completion for any section/object referring to rule names
 
-private fun collectVariantsForElement(element: PsiElement): MutableList<Pair<String, SmkRuleOrCheckpoint>> {
+private fun <Psi: SmkRuleOrCheckpoint> collectVariantsForElement(
+        element: PsiElement,
```

Comment:
`collectVariantsForElement` is private and is used only here, but has:  too generic and not type-safe signature, e.g. you could call this method with use `indexKey` for rules and pass checkpoints section class. Also its usages introduce code duplication.

Let's use smth simpler, e.g. `collectRuleorCheckpointVariants(element, isCheckPoint=True): List<Pair<String, SmkRuleOrCheckpoint>>`

Reply:
done